DIPs/DIP1003.md

```diff
@@ -0,0 +1,239 @@
+- Video games, such as referring to the player character's body
+- Working with HTTP requests and responses, which have a body
+- Working programmatically with emails, which have a body
+- Many external C and C++ libraries make use of "body" as a symbol name, making
+  porting these to D and creating bindings for them more difficult
+- Writing compilers and parsers (functions, loops, etc. all have bodies)
+- Scripting interfaces and wrappers
+
+The `body` keyword is not required from a parsing perspective; an un-annotated function block could be used
+with no difficulties. Alternatively, the `function` keyword is readily available and has the same
+meaning when used in the context of function contracts; it is very close in meaning to
+the `body` keyword in this context. The `body` keyword is in fact shorthand for the term
+"_function_ body", denoting the body of a function defintion that contains the actual code.
+Therefore, these changes will not be any less descriptive of what the keyword means in
+this context than the keyword `body`. It is anticipated that these changes may actually
+lead to _less_ confusion for programmers that are new to D.
+
+Also, the author believes that the `function` keyword is not as "overloaded" with different
+meanings in different contexts, like with other keywords such as `static`. Therefore, these
+proposed changes will not unduly "overload" the keyword `function`. There are currently 2
+other contexts in which the `function` keyword is already used:
+
+- Function variable type definitions
+- Function literal definitions
+
+Examples:
+```D
+int function(char c) fp; //Function variable type definition
+
+fp = function int(char c) { return 6; }; //Function literal definition
+
+int function() fp2 = function { return 0; }
+```
+
+If one considers that each keyword in a language entails a certain cost in terms of what names a
+user may use as a symbol, as well as the fact that new users must learn what each keyword means,
+then it follows that allowing `function` to be used in the context of function contracts has a much
+"cheaper" tradeoff than using `body`. Therefore, it makes sense logically to remove the `body`
+keyword, which has a high cost with a low return on investment, with the `function` keyword, which
+also has a relatively high cost, but a much better return on investment.
+
+### Breaking changes / deprecation process
+
+Depending on how these changes are implemented, there will be little to no breakage,
+and any breakage that _does_ occur will be done through the established deprecation
+process for breaking changes (first a warning, then deprecation, then removal).
+
+It is the author's opinion that option 4 is the ideal path forward for implementing this proposal and minimizing breakage. However, this
+may not be feasible as it requires the implementation of conditional keywords and the effort to implement them is currently unknown. Likewise,
```

Comment:
IMO conditional keywords needs a whole other DIP to address them. I'd prefer to keep this DIP focused on the issue of removing `body`.